components/emoji_picker/emoji_picker.jsx

```diff
@@ -38,358 +120,268 @@ export default class EmojiPicker extends React.Component {
             this.searchInput.focus();
         });
+        this.divHeight = this.emojiPickerContainer.offsetHeight;
     }
 
-    handleCategoryClick(category) {
-        const scrollOffset = this.categories[category].offset;
+    componentWillUpdate(nextProps, nextState) {
+        if (
+            this.lastVisibleEmoji &&
+            this.state.divTopOffset !== nextState.divTopOffset
+        ) {
+            const difference = this.lastVisibleEmoji.offsetTop - (nextState.divTopOffset + EMOJI_CONTAINER_HEIGHT + EMOJI_LAZY_LOAD_BUFFER);
+            if (difference <= 0) {
+                const numToLoad = EMOJI_TO_LOAD_PER_UPDATE + Math.ceil((difference / EMOJI_HEIGHT) * EMOJI_PER_ROW * -1);
+                this.setState((state) => ({
+                    emojisToShow: state.emojisToShow + numToLoad
+                }));
+            }
+        }
+    }
 
-        this.setState({
-            scrollOffset,
-            activeCategory: category
-        });
+    handleCategoryClick(categoryName) {
+        this.emojiPickerContainer.scrollTop = this.state.categories[categoryName].offset;
     }
 
     handleFilterChange(e) {
         e.preventDefault();
-
-        for (const category of Object.keys(this.categories)) {
-            this.categories[category].offset = 0;
-            this.categories[category].enable = false;
-        }
-
         const filter = e.target.value;
-        const list = this.generateList(filter);
-
-        let activeCategory = 'recent';
-        if (!filter) {
-            for (const category of Object.keys(this.categories)) {
-                if (this.categories[category].enable) {
-                    activeCategory = this.categories[category].name;
-                    break;
-                }
-            }
-        }
-
-        this.setState({
-            activeCategory,
-            selected: null,
-            scrollOffset: 0,
-            list,
-            filter
-        });
+        this.setState(() => ({
+            filter,
+            cursor: [0, 0]
+        }));
     }
 
-    handleItemOver(emoji) {
-        clearTimeout(this.timeouthandler);
+    handleItemOver(categoryIndex, emojiIndex) {
         this.setState({
-            selected: emoji
+            cursor: [categoryIndex, emojiIndex]
         });
     }
 
-    handleItemOut() {
-        this.timeouthandler = setTimeout(
-            () =>
-                this.setState({
-                    selected: null
-                }),
-            500
-        );
+    handleItemClick(emoji) {
+        this.props.onEmojiClick(emoji);
     }
 
-    handleItemUnmount(emoji) {
-        // Prevent emoji preview from showing emoji which is not present anymore (due to filter)
-        if (this.state.selected === emoji) {
-            this.setState({
-                selected: null
-            });
+    handleKeyDown(e) {
+        switch (e.key) {
+        case 'ArrowRight':
+            e.preventDefault();
+            this.selectNextEmoji(1);
```

Comment:
Feels weird to see a number here. Would it be better to have it with default value ? 
Like:
```
this.selectNextEmoji();
```

then with the function:
```
selectNextEmoji(offset = 1) {...}
```

Reply:
yep. I updated this. Thanks